Update fused kernels and call _safe_softmax from SDPA

drisspg opened the pull request:
Summary:
X-link: https://github.com/pytorch/pytorch/pull/131863

cc ezyang gchanan jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 Valentine233

imported-using-ghimport

Reviewed By: Chillee

Differential Revision: D61418679

Pulled By: drisspg

shoumikhin commented:
@YifanShenSZ ptal when you have a chance!

YifanShenSZ commented on backends/apple/coreml/test/test_coreml_partitioner.py:
```diff
@@ -68,15 +68,23 @@ def test_vit_skip_conv(self):
+            "getitem",
+            "aten.any.dim",
+            "executorch_call_delegate",
+        ]
```
In the image @shoumikhin showed me
![image](https://github.com/user-attachments/assets/5232e5e6-5001-4024-ba80-1d9fbd566119)
I thought each safe softmax block should be
```
"executorch_call_delegate",
"getitem",
"aten.any.dim",
"executorch_call_delegate",
"getitem",
```
?

drisspg replied:
So the script that generated this https://gist.github.com/drisspg/b4087fc8aad510926c2cbe24e89f8e3b since I was having trouble building the actual PR with executorch: https://github.com/pytorch/pytorch/pull/131863

YifanShenSZ replied:
Make sense 👍 then as long as CI green I believe this is good

drisspg committed:
Update fused kernels and call _safe_softmax from SDPA (#4772)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/4772

X-link: https://github.com/pytorch/pytorch/pull/131863

cc ezyang gchanan jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 Valentine233

imported-using-ghimport

Test Plan: Imported from OSS

Reviewed By: Chillee

Differential Revision: D61418679

Pulled By: drisspg

YifanShenSZ reviewed (approved):
Awesome safe softmax 🎊 & thanks for the subsequent fix in executorch 🙏

drisspg commented:
@shoumikhin not totally sure what to do about how red executorch trunk is on this one
